plugins/inputs/nomad/nomad.go

```diff
@@ -0,0 +1,215 @@
+package nomad
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/filter"
+	"github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type Nomad struct {
+	URL string
+
+	AuthToken       string `toml:"auth_token"`
+	AuthTokenString string `toml:"auth_token_string"`
+
+	LabelInclude []string `toml:"label_include"`
+	LabelExclude []string `toml:"label_exclude"`
+
+	labelFilter filter.Filter
+
+	ResponseTimeout config.Duration
+
+	tls.ClientConfig
+
+	RoundTripper http.RoundTripper
+}
+
+const timeLayout = "2006-01-02 15:04:05 -0700 MST"
+
+var sampleConfig = `
+  ## URL for the Nomad agent
+  url = "http://127.0.0.1:4646"
+
+  ## Use auth token for authorization. ('auth_token' takes priority)
+  ## If both of these are empty, no token will be used.
+  # auth_token = "/path/to/auth/token"
+  ## OR
+  # auth_token_string = "a1234567-40c7-9048-7bae-378687048181"
+
+  ## Labels to be added as tags. An empty array for both include and
+  ## exclude will include all labels.
+  # label_include = []
+  # label_exclude = ["*"]
+
+  ## Set response_timeout (default 5 seconds)
+  # response_timeout = "5s"
+
+  ## Optional TLS Config
+  # tls_ca = /path/to/cafile
+  # tls_cert = /path/to/certfile
+  # tls_key = /path/to/keyfile
+`
+
+func init() {
+	inputs.Add("nomad", func() telegraf.Input {
+		return &Nomad{
+			LabelInclude: []string{},
+			LabelExclude: []string{"*"},
+		}
+	})
+}
+
+func (n *Nomad) SampleConfig() string {
+	return sampleConfig
+}
+
+func (n *Nomad) Description() string {
+	return "Read metrics from the Nomad api"
+}
+
+func (n *Nomad) Init() error {
+	if n.AuthToken == "" && n.AuthTokenString == "" {
+		n.AuthToken = ""
+	}
+
+	if n.AuthToken != "" {
+		token, err := os.ReadFile(n.AuthToken)
+		if err != nil {
+			return err
+		}
+		n.AuthTokenString = strings.TrimSpace(string(token))
+	}
+
+	labelFilter, err := filter.NewIncludeExcludeFilter(n.LabelInclude, n.LabelExclude)
+	if err != nil {
+		return err
+	}
+	n.labelFilter = labelFilter
+
+	return nil
+}
+
+func (n *Nomad) Gather(acc telegraf.Accumulator) error {
+	acc.AddError(n.gatherSummary(n.URL, acc))
+	return nil
+}
```

Comment:
Why not just return the error?
```suggestion
func (n *Nomad) Gather(acc telegraf.Accumulator) error {
	return n.gatherSummary(n.URL, acc)
}
```

I even vote for merging `gatherSummary`'s code in here to save one indirection.